src/request_items.rs

```diff
@@ -289,11 +307,63 @@ impl RequestItems {
         Ok(Body::Multipart(form))
     }
 
+    fn body_from_file(self) -> Result<Body> {
+        let mut body = None;
+        if self
+            .0
+            .iter()
+            .any(|item| matches!(item, RequestItem::FormFile {key, ..} if !key.is_empty()))
+        {
+            return Err(anyhow!(
+                "Can't use file fields in JSON mode (perhaps you meant --form?)"
+            ));
+        }
+        for item in self.0 {
+            match item {
+                RequestItem::DataField(..) | RequestItem::JsonField(..) => {
+                    return Err(anyhow!(
+                        "Request body (from a file) and request data (key=value) cannot be mixed."
+                    ));
+                }
+                RequestItem::FormFile {
+                    key,
+                    file_name,
+                    file_type,
+                } => {
+                    assert!(key.is_empty());
+                    if body.is_some() {
+                        return Err(anyhow!("Can't read request from multiple files"));
+                    }
+                    body = Some(Body::File {
+                        file_type: file_type
+                            .as_deref()
+                            .or_else(|| mime_guess::from_path(&file_name).first_raw())
+                            .map(HeaderValue::from_str)
+                            .transpose()?,
+                        file_name: file_name.into(),
+                    });
+                }
+                _ => {}
+            }
+        }
+        let body = body.expect("Should have had at least one file field");
+        Ok(body)
+    }
+
+    pub fn is_multipart(&self, request_type: RequestType) -> bool {
+        match request_type {
+            RequestType::Multipart => true,
+            RequestType::Form => self.has_form_files(),
+            RequestType::Json => false,
+        }
+    }
```

Comment:
My thought in putting it here was that it's harder to forget to update it when you change `.body()`. If it's isolated in `to_curl` it might get out of sync.

Reply:
I didn't mean to completely move `is_multipart()` out from `request_items.rs`. It is just that from a glance, one could mistake `is_multipart()` for `body_as_multipart()` which is being used inside `body()`. So if we can move this function a little bit further down, I think that would help.